src/cpp/src/device_config.hpp

```diff
@@ -36,44 +39,65 @@ class DeviceConfig {
+                if (key_group_size_it != plugin_config.end()) {
+                    m_key_cache_group_size = key_group_size_it->second.as<size_t>();
                 } else {
-                    // use default f32
-                    m_kv_cache_type = ov::element::f32;
+                    m_key_cache_group_size = core.get_property(device, ov::key_cache_group_size);
+                }
+                const auto value_group_size_it = plugin_config.find(ov::value_cache_group_size.name());
+                if (value_group_size_it != plugin_config.end()) {
+                    m_value_cache_group_size = key_group_size_it->second.as<size_t>();
+                } else {
+                    m_value_cache_group_size = core.get_property(device, ov::value_cache_group_size);
+                }
+
+                // if user sets ov::kv_cache_precision hint
+                bool has_kv_cache_precision_hint = false;
+                const auto kv_cache_precision_it = plugin_config.find(ov::hint::kv_cache_precision.name());
+                if (kv_cache_precision_it != plugin_config.end()) {
+                    const auto kv_cache_precision = kv_cache_precision_it->second.as<ov::element::Type>();
+                    m_key_cache_type = m_value_cache_type = kv_cache_precision;
+                    has_kv_cache_precision_hint = true;
                 }
-            }
 
-            // if user sets ov::kv_cache_precision hint
-            const auto kv_cache_precision_it = plugin_config.find(ov::hint::kv_cache_precision.name());
-            if (kv_cache_precision_it != plugin_config.end()) {
-                const auto kv_cache_precision = kv_cache_precision_it->second.as<ov::element::Type>();
-                m_kv_cache_type = kv_cache_precision;
+                // ov::key/value_cache_precision has higher priority
+                const auto key_cache_precision_it = plugin_config.find(ov::key_cache_precision.name());
+                if (key_cache_precision_it != plugin_config.end()) {
+                    const auto key_cache_precision = key_cache_precision_it->second.as<ov::element::Type>();
+                    m_key_cache_type = key_cache_precision;
+                } else {
+                    if (!has_kv_cache_precision_hint)
+                        m_key_cache_type = core.get_property(device, ov::key_cache_precision);
+                }
+                const auto value_cache_precision_it = plugin_config.find(ov::value_cache_precision.name());
+                if (value_cache_precision_it != plugin_config.end()) {
+                    const auto value_cache_precision = value_cache_precision_it->second.as<ov::element::Type>();
+                    m_value_cache_type = value_cache_precision;
+                } else {
+                    if (!has_kv_cache_precision_hint)
+                        m_value_cache_type = core.get_property(device, ov::value_cache_precision);
+                }
```

Comment:
Most of the logic in this file should be part of CPU plugin
See CVS-161326

I suppose we only need to keep a part related to KV block size, to respect low bits formats